docs/source/developers/guide/step_by_step/set_up.rst

```diff
@@ -27,3 +27,98 @@
 ******
 Set up
 ******
+
+Install and setup Git
+=====================
+
+The Arrow project is developed using `git <https://git-scm.com/>`_
+for version control which is easily available for all common
+operating systems.
+
+You can follow the instructions to install git from GitHub
+where Arrow repository is hosted, following
+`the quickstart instructions <https://docs.github.com/en/get-started/quickstart/set-up-git>`_.
+
+When git is set up do not forget to configure your name and email
+
+.. code:: console
+
+   $ git config --global user.name "Your Name"
+   $ git config --global user.email your.email@example.com
+
+and `authenticate with GitHub <https://docs.github.com/en/get-started/quickstart/set-up-git#next-steps-authenticating-with-github-from-git>`_
+as this will allow you to interact with GitHub without typing
+a username and password each time you execute a git command.
+
+**The Terminal**
+
+You might be able to get away without using the command-line for
+most tasks, but for some it seems to be the most straightforward way.
+
+Get the source code
+===================
+
+Fork the repository
+-------------------
+
+The Arrow GitHub repository contains both the Arrow C++ library and
+the various bindings to it (libraries in Python, R, Rust, Java, etc.).
```

Comment:
Well, except that Rust, Java and other implementations (such as C#, Go or Javascript) are not bindings to Arrow C++; besides, the Rust implementation is in a separate repository ;-)